crates/transaction-pool/src/maintain.rs

```diff
@@ -571,70 +573,76 @@ fn changed_accounts_iter(
+        })
+        .collect::<Vec<_>>();
+    let _ = pool.add_transactions(crate::TransactionOrigin::Local, pool_transactions).await;
+    reth_primitives::fs::remove_file(file_path.clone())?;
+    Ok(())
 }
 
-fn save_local_txs_backup<P>(pool: P, transactions_path: Option<PathBuf>)
+fn save_local_txs_backup<P>(pool: P, file_path: PathBuf)
 where
     P: TransactionPool + TransactionPoolExt + 'static,
 {
-    if let Some(file_path) = transactions_path {
-        let local_transactions = pool.get_local_transactions();
-        let local_transactions = local_transactions
-            .into_iter()
-            .map(|tx| tx.to_recovered_transaction().into_signed())
-            .collect::<Vec<_>>();
-
-        let num_txs = local_transactions.len();
-        let mut buf = alloy_rlp::BytesMut::new();
-        alloy_rlp::encode_list(&local_transactions, &mut buf);
-        trace!(target: "reth::cli", txs_file =?file_path, num_txs=%num_txs, "Saving current local transactions");
-        let parent_dir = file_path.parent().map(std::fs::create_dir_all).transpose();
-        match parent_dir.map(|_| reth_primitives::fs::write(&file_path, buf)) {
-            Ok(_) => {
-                info!(target: "reth::cli", txs_file=?file_path, "Wrote local transactions to file");
-            }
-            Err(err) => {
-                warn!(target: "reth::cli", ?err, txs_file=?file_path, "Failed to write local transactions to file");
-            }
+    let local_transactions = pool.get_local_transactions();
+    if local_transactions.is_empty() {
+        info!("no local transactions to recover");
+        return
+    }
+
+    let local_transactions = local_transactions
+        .into_iter()
+        .map(|tx| tx.to_recovered_transaction().into_signed())
+        .collect::<Vec<_>>();
+
+    let num_txs = local_transactions.len();
+    let mut buf = alloy_rlp::BytesMut::new();
+    alloy_rlp::encode_list(&local_transactions, &mut buf);
+    info!(target: "reth::cli", txs_file =?file_path, num_txs=%num_txs, "Saving current local transactions");
+    let parent_dir = file_path.parent().map(std::fs::create_dir_all).transpose();
+    match parent_dir.map(|_| reth_primitives::fs::write(&file_path, buf)) {
+        Ok(_) => {
+            info!(target: "reth::cli", txs_file=?file_path, "Wrote local transactions to file");
+        }
+        Err(err) => {
+            warn!(target: "reth::cli", ?err, txs_file=?file_path, "Failed to write local transactions to file");
         }
     }
 }
 
+enum ApplyLocalTxsBackupError {
+    DecodeError(alloy_rlp::Error),
+    FsPathError(FsPathError),
+}
+
+impl From<FsPathError> for ApplyLocalTxsBackupError {
+    fn from(err: FsPathError) -> Self {
+        ApplyLocalTxsBackupError::FsPathError(err)
+    }
+}
+
+impl From<alloy_rlp::Error> for ApplyLocalTxsBackupError {
+    fn from(err: alloy_rlp::Error) -> Self {
+        ApplyLocalTxsBackupError::DecodeError(err)
+    }
+}
```

Comment:
this should use thiserror::Error

and use the #[from] attributes,
then the error messages of L660 can be moved into the error attribute